### PR TITLE
Code Insights landing page and other updates

### LINKED ISSUE #5099: Code Insights Landing Page & Site Updates
NOTE: There is a press embargo on code insights so all design, development and reviews will need to be done locally and privately.

More details including design and copy will be added here and status will be moved to ready for development once all requirements are gathered.

In addition to the landing page, we will be updating copy on a few parts of the site:

1. [Home Page](https://about.sourcegraph.com/)
2. [/pricing](https://about.sourcegraph.com/pricing/)
3. [/news](https://about.sourcegraph.com/news/)
4. [/press-release](https://about.sourcegraph.com/press-release/)

Copy doc (WIP) for these changes [here](https://docs.google.com/document/d/1JnXDCnaq981QPZziZZj2zMyihfojuC65gnBYqrn3SSY/edit).

🔗[Meta](https://docs.google.com/spreadsheets/d/11DYIN1e6-weZ0Rb9-_bmnx1Uje8fs2mVoQUUdjjYBqk/edit#gid=572619067)

### PR DESCRIPTION
**Pull request by katjuell**
Adds Code Insights landing page, along with changes to the homepage and the pricing page. Also adds a contact form for demos. Closes #5099 and #5141 

For a list of copy changes, see [this doc](https://docs.google.com/document/d/1JnXDCnaq981QPZziZZj2zMyihfojuC65gnBYqrn3SSY/edit), [this doc](https://docs.google.com/document/d/1oJBljDO-u_kuhPY2gDM9iNYeIFWdVmCUsOUeM83_vf0/edit) and the [Figma](https://www.figma.com/file/BWNiKcVBcRBQSS1VtG47hl/Code-Insights?node-id=257%3A8067).

## To Test
- QA homepage changes on desktop/ mobile
- QA pricing page on desktop/ mobile
- QA code insights page on desktop/ mobile

### CODE REVIEW

File: website/package.json
Comment (katjuell) [inline website/package.json:74]: Not sure why we don't have these here, so I added them.

File: website/src/pages/code-insights.tsx
Comment (katjuell) [inline website/src/pages/code-insights.tsx:244]: These will be updated when we have the correct assets and blog posts to link to.

File: website/src/pages/code-insights.tsx
Comment (katjuell) [inline website/src/pages/code-insights.tsx:53]: Do these need to be updated? Going off what's in the Figma

File: website/src/pages/code-insights.tsx
Comment (katjuell) [inline website/src/pages/code-insights.tsx:428]: Waiting on video — using placeholder for now.

File: website/src/pages/code-insights.tsx
Comment (katjuell) [inline website/src/pages/code-insights.tsx:22]: There will be more code insights charts as of Monday the 21st.

File: website/src/pages/index.tsx
Comment (katjuell) [inline website/src/pages/index.tsx:258]: Also a placeholder until we get the video.

File: website/src/components/CustomCarousel.tsx
Comment (katjuell) [inline website/src/components/CustomCarousel.tsx:48]: We are going to update the toggle here to be on opacity, so the "light" arrow shows up on carousels that have gradient backgrounds (like on `/cloud-beta`). I will update this.

File: website/src/pages/contact/request-code-insights-demo.tsx
Comment (marybelzer) [reply website/src/pages/contact/request-code-insights-demo.tsx:12]: @katjuell confirming the `portalId` and `formId` are correct, those were in the embed code marketing provided. Still an open question about targetId.

Comment (marybelzer) [reply website/src/pages/contact/request-code-insights-demo.tsx:12]: @katjuell Marketing doesn't know what that targetId should be, and that hubspot embed code they gave us didn't include one. Is it required?

Comment (katjuell) [reply website/src/pages/contact/request-code-insights-demo.tsx:12]: @marybelzer that is a required field, yeah. Is there any harm in keeping it as it is? Doesn't seem like there would be?

File: website/src/components/CustomCarousel.tsx
Comment (katjuell) [reply website/src/components/CustomCarousel.tsx]: Done!

**Comment by stephzabala** [comment]
[Low priority] Padding between subtext ("Transform your...") and CTA is 40px, while padding between Headline and subtext is 25px. (Visually the goal is for the spacing to look even above and below the subtext)
 
Screen Shot 2022-02-18 at 1 58 15 PM

**Comment by stephzabala** [comment]
Ideally this quote width (title, body copy, all of it) is 5 columns wide (in my designs). Not sure how this translates to the design system that the About site is using, but TLDR can we reduce the width of the text container/increase the left and right padding?

Intended

**Comment by stephzabala** [comment]
[Medium Priority] I am using the [Phosphor Icons plug-in](https://www.figma.com/community/file/903830135544202908) for Figma for these icons (in "light"). We used this icon set for the careers page as well. But if there is an icon library you've found for the rest of the About site lmk! It seemed like we weren't using one before.

Screen Shot 2022-02-18 at 2 32 24 PM

**Comment by stephzabala** [comment]
[Low priority] Across the whole page design, I am using only the Semi Bold weight of Source Sans for headers. This is a huge inconsistency across the site (we seem to not have rules for when to use bold/semi-bold) and should be addressed in the new DLS conversation... but for consistency I am just using semi-bold here.

**Comment by katjuell** [comment]
> [Low priority] Padding between subtext ("Transform your...") and CTA is 40px, while padding between Headline and subtext is 25px. (Visually the goal is for the spacing to look even above and below the subtext)
> 
> Screen Shot 2022-02-18 at 1 58 15 PM

Hey @stephzabala, Bootstrap is not giving me the exact 40px. Let me know if you want me to override that with a custom class. The choices were between `32px` and `64px`, so I went with the larger, but that may be too much and a custom class may be the way to go.
Screen Shot 2022-02-18 at 2 50 18 PM

File: website/src/css/pages/_code-insights.scss
Comment (bretthayes) [reply website/src/css/pages/_code-insights.scss:57]: It might be a good idea for UX to prevent vertical scroll content jumps when different slides are changed. Or rather, move the arrows above the content (second screenshot) so that you don't notice the jump since the content would be below. But we should stack the arrow/dots UI and the graphs/content up to 990px breakpoint since the text gets pretty crammed between 768–990px (first screenshot).

Screen Shot 2022-02-18 at 3 00 48 PM

Feel free to add input via the suggestion of moving the arrows to the top for better UX (below screenshot) @stephzabala 

Screen Shot 2022-02-18 at 3 02 00 PM

File: website/src/css/pages/_code-insights.scss
Comment (bretthayes) [inline website/src/css/pages/_code-insights.scss:9]: I'm seeing the hero cut off after around 2045px. I'm not sure if adding `background-size: cover;` would suffice but we should span the background across the entire width.

Screen Shot 2022-02-18 at 1 35 06 PM

**Review by bretthayes — CHANGES_REQUESTED** [review]

File: website/src/components/CustomCarousel.tsx
Comment (katjuell) [reply website/src/components/CustomCarousel.tsx:3]: Looks like we [get it as a dependency](https://github.com/sourcegraph/about/blob/2eee4b9ffa8b4b2dfa9f787e362f7da269e5c4ae/website/yarn.lock#L2541) for a bunch of the packages that got added with the Code Insights custom components. I'm 100% for using `clsx` for the replatform, but for now & for this my vote would be to go with what we already have.

File: website/src/css/pages/_code-insights.scss
Comment (katjuell) [reply website/src/css/pages/_code-insights.scss:9]: 100% yay!

File: website/src/pages/contact/request-code-insights-demo.tsx
Comment (marybelzer) [reply website/src/pages/contact/request-code-insights-demo.tsx:12]: @katjuell I don't think so, Marketing said as far as they know they have not included that in the past, so I'm wondering if Mate just put something that kind of made sense?

Comment (katjuell) [reply website/src/pages/contact/request-code-insights-demo.tsx]: @marybelzer you were right! Turns out it's just the `target` from HubSpot's API renamed slightly. Updated it in our form!

**Comment by marybelzer** [comment]
@katjuell Alicja noticed the illustration is not placed correctly - alternative export that should work better attached, she'll review again once we have the new illustration in
Illustration export

**Review by bretthayes — CHANGES_REQUESTED** [review]
Nothing major or blocking, just some improvements. Otherwise looks great!

File: website/src/components/Header.tsx
Comment (katjuell) [reply website/src/components/Header.tsx:187]: Agreed! However, it is a larger cleanup job to deal with all of these right now, pre-launch, and I would rather do them all at once to achieve consistency across the site. I have created a [cleanup issue](https://github.com/sourcegraph/about/issues/5164), which I will address post-launch.

File: website/src/pages/index.tsx
Comment (katjuell) [reply website/src/pages/index.tsx:278]: Will be covered by [this cleanup issue](https://github.com/sourcegraph/about/issues/5164)

File: website/src/components/Header.tsx
Comment (katjuell) [reply website/src/components/Header.tsx:187]: I think, given the circumstances, we can now roll this in. Do you have any objections to rolling that issue into this PR?

Comment (katjuell) [reply website/src/components/Header.tsx:187]: Thinking about this more, I think the cleanup should be a separate PR. I will work on that and we can merge it before we merge this one.

Comment (bretthayes) [reply website/src/components/Header.tsx:187]: Yeah, I mean, we could do an incremental approach and just do the Link elements in this PR since there are only a few. Then at least this page is improved. But for a larger scoped ticket like #5164, since we're re-platforming, those will be replaced with NextJS Link elements anyway. So the question is; do we even bother doing the entire site right now?

Comment (katjuell) [reply website/src/components/Header.tsx:187]: That is a good question! What if we split the difference and scope #5164 to be "Use Link elements for internal links on nav and homepage"? Then we at least make improvements to the entirety of the page/ component we are touching in this PR

Comment (bretthayes) [reply website/src/components/Header.tsx:187]: That sounds good to me! Let's do that and split the difference.

Comment (katjuell) [reply website/src/components/Header.tsx:187]: And then I will update these here

File: website/src/components/Header.tsx
Comment (bretthayes) [reply website/src/components/Header.tsx:68]: Yes, I think just on the homepage but we should confirm with a stakeholder

**Review by bretthayes — APPROVED** [review]

File: website/src/components/Header.tsx
Comment (katjuell) [reply website/src/components/Header.tsx:68]: cc @marybelzer

**Comment by katjuell** [comment]
@marybelzer just wanted to confirm that we are only showing the CTA banner on the homepage?

**Comment by marybelzer** [comment]
@katjuell Yes, just on the homepage

**Comment by katjuell** [comment]
Awesome — tyvm!

**Comment by felixfbecker** [comment]
Heads up, this PR needs one approval before it can be merged and hasn't been approved yet. It needs to be approved before we can merge it tomorrow morning.

File: website/src/pages/code-insights.tsx
Comment (katjuell) [reply website/src/pages/code-insights.tsx]: Nope! Removed.

**Review by bretthayes — APPROVED** [review]

**Pull request merged**
